registry/storage/driver/base/base.go

```diff
@@ -50,16 +50,40 @@ type Base struct {
 	storagedriver.StorageDriver
 }
 
+// Format errors received from the storage driver
+func (base *Base) setDriverName(e error) error {
+	if e != nil {
+		if actualErr, ok := e.(storagedriver.ErrUnsupportedMethod); ok {
```

Comment:
It might be more extensible to use `reflect` to check if the struct has a `DriverName` field and set it appropriately.